File: ZLGitHubClient/ZLGitHubClient/Class/Base/ZLUtils/ZLToastView/ZLToastView.m
//
//  ZLToastView.m
//  ZLGitHubClient
//
//  Created by 朱猛 on 2020/1/19.
//  Copyright © 2020 ZM. All rights reserved.
//

#import "ZLToastView.h"
#import <Toast/Toast.h>

@implementation ZLToastView

+ (void) showMessage:(NSString *) message
{
    UIWindow * keyWindow = [UIApplication sharedApplication].keyWindow;
    [keyWindow makeToast:message];
}

+ (void) showMessage:(NSString *)message duration:(NSTimeInterval) duration;
{
    UIWindow * keyWindow = [UIApplication sharedApplication].keyWindow;
    [keyWindow makeToast:message duration:duration position:CSToastPositionBottom];
}

+ (void) showMessage:(NSString *)message duration:(NSTimeInterval) duration sourceView:(UIView *) view
{
    [view makeToast:message duration:duration position:CSToastPositionBottom];
}

@end

File: ZLGitHubClient/ZLGitHubClient/Class/Utility/ZLUtils/ZLToastView/ZLToastView.swift
//
//  ZLToastView.swift
//  ZLGitHubClient
//
//  Created by 朱猛 on 2022/3/27.
//  Copyright © 2022 ZM. All rights reserved.
//

import UIKit
import Toast

@objcMembers class ZLToastView: NSObject {
    
    class func showMessage(_ message: String) {
        let window = UIApplication.shared.keyWindow
        window?.makeToast(message, duration: 3.0, position: CSToastPositionCenter)
    }
    
    class func showMessage(_ message: String, duration: TimeInterval) {
        let window = UIApplication.shared.keyWindow
        window?.makeToast(message, duration: duration, position: CSToastPositionCenter)
    }
    
    class func showMessage(_ message: String, duration: TimeInterval, sourceView: UIView) {
        sourceView.makeToast(message, duration: duration, position: CSToastPositionCenter)
    }
    
    class func showMessage(_ message: String, sourceView: UIView) {
        sourceView.makeToast(message, duration: 3.0, position: CSToastPositionCenter)
    }

}

